### PR TITLE
Built: refactor tasks to file, demo template

### PR DESCRIPTION
**Pull request by folletto**
screen shot 2017-07-21 at 16 29 35

This PR:

* moves all the custom tasks to individual files so they are more easily reused and updated
* moves the template for the demo outside the Grunt file in its own template file

Why?

1. Easier to spot what changed in the build process
2. More easily reused across different repositories (Gridicons, Social Icons, Dashicons)
3. Cleaner separation of dependencies
4. Folder and files input declared in config instead of being hardcoded
5. Separate demo template
6. Separate react template

Note: there's also one icon added as PDF as it was added before we had the automated PDF build introduced in #238. 

### How to test

1. Switch to this branch
2. `grunt`
3. Check no changes to any built file anywhere, with the exception of the tabulation of the demo.
4. Check the compiled React component works.

### CODE REVIEW

**Comment by folletto** [comment]
FWIW, I ran a diff for `index.jsx` and `example.jsx` and they match.

**Comment by drw158** [comment]
everything seems to work 👍

**Comment by gwwar** [comment]
Did we forget to check in pdf/gridicons-nametag.pdf ? I see this as modified when I run npm install.

**Comment by folletto** [comment]
Yes: 704839da512f4003457249af9c346af8abb7359e

**Comment by gwwar** [comment]
For folks 👀 you don't need a global grunt installed, the equivalent script that uses grunt from node_modules is `npm run build`

File: grunt-tasks/svg-to-pdf.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        pdf.info.CreationDate = "";
+        pdf.pipe( fs.createWriteStream( files.dest + svgFile.slice(0, -4) + '.pdf' ) );
+        pdf.on('close', function() { done(); });
+
+        svg2pdfkit(pdf, fileContent, 0, 0 );
+        pdf.end();
+      } )
+    } );
+  } );
+}
```
Comment (gwwar) [inline grunt-tasks/svg-to-pdf.js]: Don't forget the `;` for assignments. 😄 Looks like the other split up scripts need one too.

**Comment by folletto** [comment]
Ok done:

1. Updated to master
2. Cleaned up the missing ";".

**Comment by gwwar** [comment]
@folletto would you be interested in also updating the social-logos build? We'd be able to get rid of the handlebars dependency then

**Comment by folletto** [comment]
Ok, added to my todo list :)

**Comment by folletto** [comment]
Refactoring of social logos build PR: https://github.com/Automattic/social-logos/pull/58

**Pull request merged**